src/integration.ts

```diff
@@ -2,29 +2,48 @@ import { ApiGatewayVtlIntegration } from "./api";
 import { AppSyncVtlIntegration } from "./appsync";
 import { ASL, State } from "./asl";
 import { EventBus, EventBusTargetIntegration } from "./event-bridge";
-import { CallExpr } from "./expression";
+import { AwaitExpr, CallExpr, PromiseExpr, ReferenceExpr } from "./expression";
 import { Function, NativeIntegration } from "./function";
 import {
+  isAwaitExpr,
   isCallExpr,
-  isExprStmt,
-  isIdentifier,
-  isPropAccessExpr,
+  isPromiseExpr,
   isReferenceExpr,
-  isReturnStmt,
-  isVariableStmt,
 } from "./guards";
 import { FunctionlessNode } from "./node";
 import { AnyFunction } from "./util";
+import { visitEachChild } from "./visit";
 import { VTL } from "./vtl";
 
 export const isIntegration = <I extends IntegrationInput<string, AnyFunction>>(
   i: any
 ): i is I => typeof i === "object" && "kind" in i;
 
-export function isIntegrationCall(node: FunctionlessNode): node is CallExpr {
+export type IntegrationCallExpr = CallExpr & { expr: ReferenceExpr };
+
+export function isIntegrationCallExpr(
+  node: FunctionlessNode
+): node is IntegrationCallExpr {
   return isCallExpr(node) && isReferenceExpr(node.expr);
 }
 
+export function isIntegrationCallPattern(
+  node: FunctionlessNode
+): node is
+  | IntegrationCallExpr
+  | (AwaitExpr & { expr: IntegrationCallExpr })
+  | (PromiseExpr & { expr: IntegrationCallExpr })
+  | (AwaitExpr & { expr: PromiseExpr & { expr: IntegrationCallExpr } }) {
```

Comment:
I'm curious - can this type be inferred from the impl?

Reply:
No, because the user can choose the pattern. Await may be on a non-promise, a promise may be on a return, the integration may not return a promise.

Reply:
Oh I meant the return type signature

Reply:
Actually updated this to a type now so we can use it for other things.